src/main/java/uk/ac/ebi/atlas/hcalandingpage/HcaMetadataService.java

```diff
@@ -0,0 +1,62 @@
+package uk.ac.ebi.atlas.hcalandingpage;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import org.springframework.stereotype.Component;
+import uk.ac.ebi.atlas.model.experiment.Experiment;
+import uk.ac.ebi.atlas.trader.ExperimentTrader;
+
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+
+@Component
+public class HcaMetadataService {
+    private final HcaMetadataDao hcaMetadataDao;
+    private final ExperimentTrader experimentTrader;
+
+    public HcaMetadataService(ExperimentTrader experimentTrader,
+                              HcaMetadataDao hcaMetadataDao) {
+        this.experimentTrader = experimentTrader;
+        this.hcaMetadataDao = hcaMetadataDao;
+    }
+
+    public ImmutableSet<String> getHcaOntologyIds() {
+        return getMetadata().get("ontology_ids");
+    }
+
+    public ImmutableSet<Experiment> getHcaExperiments() {
+        return getHcaExperiments(getMetadata().get("experiment_accessions"));
+    }
+
+    private ImmutableMap<String, ImmutableSet<String>> getMetadata() {
+        var humanExperimentAccessionsAndAssociatedOntologyIds =
+                hcaMetadataDao.fetchHumanExperimentAccessionsAndAssociatedOntologyIds();
+
+        var ontologyIds = humanExperimentAccessionsAndAssociatedOntologyIds
+                .stream()
+                .flatMap(metadataList -> metadataList
+                        .stream()
+                        .map(metadataMap -> {
+                            var url = metadataMap.get("ontology_annotation").toString();
+                            return url.substring(url.lastIndexOf('/') + 1);
+                        }))
+                .collect(toImmutableSet());
+
+        var experimentAccessions = humanExperimentAccessionsAndAssociatedOntologyIds
+                .stream()
+                .flatMap(metadataList -> metadataList
+                        .stream()
+                        .map(metadataMap -> metadataMap.get("experiment_accession").toString()))
+                .collect(toImmutableSet());
+
+        return ImmutableMap.of(
+                "ontology_ids", ontologyIds,
+                "experiment_accessions", experimentAccessions);
+    }
+
+    private ImmutableSet<Experiment> getHcaExperiments(ImmutableSet<String> experimentAccessions) {
+        return experimentAccessions
+                .stream()
+                .map(experimentTrader::getPublicExperiment)
+                .collect(toImmutableSet());
+    }
```

Comment:
These two private methods add nothing of value to the class. Move the relevant blocks to the two public methods. Also note that you’re creating a map for no actual reason.

Reply:
Agreed! Have moved logic to public methods. Code is much easy to follow now.